client/src/client_error.rs

```diff
@@ -69,19 +71,34 @@ impl From<ClientErrorKind> for TransportError {
             ClientErrorKind::SigningError(err) => Self::Custom(format!("{:?}", err)),
             ClientErrorKind::FaucetError(err) => Self::Custom(format!("{:?}", err)),
             ClientErrorKind::Custom(err) => Self::Custom(format!("{:?}", err)),
+            ClientErrorKind::QuinnError(err) => Self::Custom(format!("{:?}", err)),
         }
     }
 }
 
+#[derive(Error, Debug)]
+pub enum QuinnError {
```

Comment:
I'd prefer something like the following:

```
#[derive(Error, Debug)]
pub enum QuicError {
    #[error(transparent)]
    WriteError(#[from] WriteError),
    #[error(transparent)]
    ConnectionError(#[from] ConnectionError),
    #[error(transparent)]
    ConnectError(#[from] ConnectError),
}

impl From<QuicError> for ClientErrorKind {
    fn from(quic_error: QuicError) -> Self {
        Self::Custom(format!("{:?}", quic_error))
    }
}
```
By doing this, it should be possible to get rid of `impl From<WriteError> for ClientErrorKind` `impl From<ConnectError> for ClientErrorKind` and `impl From<ConnectionError> for ClientErrorKind`. As well, I think this name is better since the name QuicError makes no reference to the specific quic library we're using. `ClientErrorKind::QuinnError(err) => Self::Custom(format!("{:?}", err))` may also be removed.